### PR TITLE
cli/named_args: refactor if statements

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
refactor if-statements in `def resolve_keg(name)`  because it was hard to read

### CODE REVIEW

File: Library/Homebrew/cli/named_args.rb [hunk 1]
```diff
@@ -265,28 +265,20 @@ def resolve_keg(name)
...
+          return Keg.new(linked_keg_ref.resolved_path) if linked_keg_ref.symlink? && linked_keg_ref.directory?
+          return Keg.new(dirs.first) if dirs.length == 1
 
-            unless (prefix = f.latest_installed_prefix).directory?
-              raise MultipleVersionsInstalledError, <<~EOS
-                #{rack.basename} has multiple installed versions
-                Run `brew uninstall --force #{rack.basename}` to remove all versions.
-              EOS
-            end
+          f = (name.include?("/") || File.exist?(name)) ? Formulary.factory(name) : Formulary.from_rack(rack)
```
Comment (Rylan12) [inline Library/Homebrew/cli/named_args.rb:272]:
```suggestion
          f = if name.include?("/") || File.exist?(name)
            Formulary.factory(name)
          else
            Formulary.from_rack(rack)
          end
```

I think I prefer this to a ternary

**Review by Rylan12 — COMMENTED** [review]
Looks good! Nice cleanup, @hyuraku!

Just one suggestion. I, personally, like to read multiline if statements over long ternary operators, so I think using one here increases readability even more. Thoughts?

File: Library/Homebrew/cli/named_args.rb (hunk 1 above)
Comment (hyuraku) [reply Library/Homebrew/cli/named_args.rb:272]: > I think I prefer this to a ternary

I agree. On second thought, this ternary is long to read.
I'll fix it soon.

**Review by Rylan12 — APPROVED** [review]
Thanks, @hyuraku! LGTM

**Comment by MikeMcQuaid** [comment]
Much nicer thanks @hyuraku!

**Pull request merged**